features/jetpack-licensing.php

```diff
@@ -27,6 +27,112 @@ function sanitize_products( $products ) {
 	return $products;
 }
 
+/**
+ * Get all Jetpack product families that we can issue licenses for from the API.
+ *
+ * @return array|WP_Error The available products.
+ */
+function fetch_product_families() {
+	$response = wp_remote_get( 'https://public-api.wordpress.com/wpcom/v2/jetpack-licensing/product-families' );
+	$status   = wp_remote_retrieve_response_code( $response );
+	$body     = wp_remote_retrieve_body( $response );
+
+	if ( 200 !== $status ) {
+		return new WP_Error(
+			'get_products_request_failed',
+			// Translators: %s = error message returned by an API
+			sprintf( __( 'Get products request failed with: %s', 'jurassic-ninja' ), $body ),
+			[
+				'status' => $status,
+			]
+		);
+	}
+
+	return json_decode( $body );
+}
+
+/**
+ * Get all Jetpack product families that we can issue licenses for.
+ *
+ * @return array The available products.
+ */
+function get_product_families() {
+	$cache_key = 'jurassic_ninja:jetpack_licensing_products';
+	$cache     = get_transient( $cache_key );
+
+	if ( $cache ) {
+		return $cache;
+	}
+
+	$families = fetch_product_families();
+
+	if ( is_wp_error( $families ) || empty( $families ) ) {
```

Comment:
I don't know how I feel about hardcoding the failover object here. I think we can go around this by setting the transient with a _very_ long expiration on every (or a subset of every) successful call and showing it if the call fails. That way we can get a simple failover and we don't have to hard code anything. Judging from the `jurassic-ninja/site-launched` stats  in MC this many transients writes would not even move the needle.

Reply:
Switched to request every time and save the last successful call as a fallback. It is not expected for this page to get any kind of high traffic but we want it to be as fresh as possible so this should be better.